plex_generate_previews.py

```diff
@@ -8,23 +9,84 @@
 import glob
 import os
 import struct
-import gpustat
+
+if not os.path.isfile('/usr/bin/mediainfo'):
+    print('MediaInfo not found.  MediaInfo must be installed and available in PATH.')
+    sys.exit()	
+
+try:
+    from pymediainfo import MediaInfo
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install pymediainfo".')
+    sys.exit()
+
+try:
+    import gpustat
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install gpustat".')
+    sys.exit()
+ 
 import time
-import requests
+
+try:
+    import requests
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install requests".')
+    sys.exit()
+
 import array
-from plexapi.server import PlexServer
-from loguru import logger
-from rich.console import Console
-from rich.progress import Progress, SpinnerColumn, MofNCompleteColumn
+
+try:
+    from plexapi.server import PlexServer
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install plexapi".')
+    sys.exit()
+    
+try:
+    from loguru import logger
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install loguru".')
+    sys.exit()
+
+try:
+    from rich.console import Console
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install rich".')
+    sys.exit()
+
+try:
+    from rich.progress import Progress, SpinnerColumn, MofNCompleteColumn
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install rich".')
+    sys.exit()
+
+try:
+    from PIL import Image
+except ImportError:
+    print('Dependencies Missing!  Please run "pip3 install pillow".')
+    sys.exit()
+
+import PIL
+
+if not os.path.isfile('/usr/bin/ffmpeg'):
+    print('FFmpeg not found.  FFmpeg must be installed and available in PATH.')
+    sys.exit()	
 
 # EDIT These Vars #
 PLEX_URL = 'https://xxxxxx.plex.direct:32400/'
 PLEX_TOKEN = 'xxxxxx'
-PLEX_BIF_FRAME_INTERVAL = 2
+PLEX_BIF_FRAME_INTERVAL = 5
 PLEX_LOCAL_MEDIA_PATH = '/path_to/plex/Library/Application Support/Plex Media Server/Media'
-TMP_FOLDER = '/tmp/plex'
+TMP_FOLDER = '/dev/shm/plex_generate_previews'
 GPU_THREADS = 4
 CPU_THREADS = 4
+PLEX_UID = 1000
+PLEX_GUID = 1001
+
+# DO NOT EDIT BELOW HERE #
+
+os.setgid(PLEX_GUID)
```

Comment:
@jasonsansone won't this fail unless it is running as root? wouldn't the better option to tell the user the run the script as that user?